fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckMessageCard.kt

```diff
@@ -0,0 +1,215 @@
+                    onClick = onButtonClick,
+                )
+            }
+        }
+    }
+}
+
+/**
+ * The possible types of a [ReviewQualityCheckMessageCard].
+ */
+enum class ReviewQualityCheckMessageType {
+
+    Warning,
+    Confirmation,
+    Error,
+    Info,
+    ;
+
+    val cardBackgroundColor: Color
+        @Composable
+        get() = when (this) {
+            Warning -> FirefoxTheme.colors.layerWarning
+            Confirmation -> FirefoxTheme.colors.layerConfirmation
+            Error -> FirefoxTheme.colors.layerError
+            Info -> FirefoxTheme.colors.layerInfo
+        }
+
+    val buttonBackgroundColor: Color
+        @Composable
+        get() = when (this) {
+            Warning -> FirefoxTheme.colors.actionWarning
+            Confirmation -> FirefoxTheme.colors.actionConfirmation
+            Error -> FirefoxTheme.colors.actionError
+            Info -> FirefoxTheme.colors.actionInfo
+        }
+
+    val buttonTextColor: Color
+        @Composable
+        get() = when {
+            this == Info && !isSystemInDarkTheme() -> FirefoxTheme.colors.textOnColorPrimary
+            else -> FirefoxTheme.colors.textPrimary
+        }
+}
+
+private class PreviewModel(
+    val messageType: ReviewQualityCheckMessageType,
+    val iconId: Int,
+)
+
+private class PreviewModelParameterProvider : PreviewParameterProvider<PreviewModel> {
+    override val values: Sequence<PreviewModel>
+        get() = sequenceOf(
+            PreviewModel(ReviewQualityCheckMessageType.Warning, R.drawable.mozac_ic_warning_fill_24),
+            PreviewModel(ReviewQualityCheckMessageType.Confirmation, R.drawable.mozac_ic_checkmark_24),
+            PreviewModel(ReviewQualityCheckMessageType.Error, R.drawable.mozac_ic_information_fill_24),
+            PreviewModel(ReviewQualityCheckMessageType.Info, R.drawable.mozac_ic_information_fill_24),
+        )
+}
+
+@LightDarkPreview
+@Composable
+private fun StatusCardPreview(
+    @PreviewParameter(PreviewModelParameterProvider::class) model: PreviewModel,
+) {
+    FirefoxTheme {
+        Box(
+            modifier = Modifier
+                .background(color = FirefoxTheme.colors.layer1)
+                .padding(16.dp),
+        ) {
+            ReviewQualityCheckMessageCard(
+                title = "Title text",
+                messageType = model.messageType,
+                modifier = Modifier.fillMaxWidth(),
+                description = "Description text",
+                linkText = "Primary link text with an underlined hyperlink.",
+                hyperlinkText = "underlined hyperlink",
+                buttonText = "Button text",
+                onLinkClick = {},
+                onButtonClick = {},
+                icon = {
+                    Icon(
+                        painter = painterResource(id = model.iconId),
+                        contentDescription = null,
+                        modifier = Modifier.size(24.dp),
```

Comment:
Not sure if we really need to set the size. I believe `Icon` can determine the size based on the provided drawable.

Reply:
I needed to set it here because one of the icons is named 24, but actually isn't 24